clang/test/Driver/sycl-offload.c

```diff
@@ -1142,3 +1142,8 @@
 // RUN:   | FileCheck -check-prefixes=CHK-FSYNTAX-ONLY,CHK-NO-EMIT-IR %s
 // CHK-FSYNTAX-ONLY: clang{{.*}} "-cc1" "-triple" "spir64-unknown-unknown"{{.*}} "-fsyntax-only"
 // CHK-NO-EMIT-IR-NOT: "-emit-llvm-bc"
+
+/// ###########################################################################
+/// Verify that -save-temps puts header/footer in a correct place
+// RUN: %clang -fsycl -fno-sycl-device-lib=all -fsycl-targets=spir64-unknown-unknown -target x86_64-unknown-linux-gnu -save-temps %s -### 2>&1 | FileCheck %s -check-prefixes=CHECK-SAVE-TEMPS-DIR
+// CHECK-SAVE-TEMPS-DIR: clang{{.*}} "-fsycl-int-header=./sycl-offload-header-{{[a-z0-9]*}}.h"{{.*}}"-fsycl-int-footer=./sycl-offload-footer-{{[a-z0-9]*}}.h"
```

Comment:
Can you also check behaviors with `-save-temps=obj` and `-o`?

Reply:
Done

Reply:
Interestingly enough, `-save-temps=obj` with `-o` containing a relative path breaks the compilation. It fails to correctly construct the output file names for `sycl-post-link` job. As that is also present in main branch, I'd like to address it in a separate PR.